localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -264,7 +264,10 @@ def get_records(self, shard_iterator: str) -> dict:
             )
             return get_records_response
         # TODO: test iterator expired with conditional error scenario (requires failure destinations)
-        except self.source_client.exceptions.ExpiredIteratorException as e:
+        except (
+            self.source_client.exceptions.ExpiredIteratorException,
+            self.source_client.exceptions.TrimmedDataAccessException,
```

Comment:
Ah, that's nice, I thought we'd need to do string matching again, but both `dynamodbstreams` and the `kinesis` clients implements the `TrimmedDataAccessException`? That's practical 😄

Reply:
Kinesis doesn't raise this ://// will re-implement